lib/fog/storage/google_json/requests/put_object.rb

```diff
@@ -46,6 +46,8 @@ def put_object(bucket_name,
                        kms_key_name: nil,
                        predefined_acl: nil,
                        **options)
+          data = StringIO.new(data) if options[:content_type] == "text/plain"
```

Comment:
Instead of this, could you change this to something like: ```
if data.is_a?(String)
  data = StringIO.new(data)
end
```

Because its not the content type, its the data type being passed in thats an issue.

Reply:
I wonder, in application design, file paths are considered as an argument of `data`.

According to the comment, it only assumes stream/raw content/paperclip thing. So maybe we can wrap in a stream strightforward.

> `# @param data [File|String|Paperclip::AbstractAdapter] File, String or Paperclip adapter to create object from`

I'm wondering this because `insert_object` considers `data` as stream or file path.